src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private int size;
+    private Node<K, V>[] storage;
+
+    public MyHashMap() {
+        storage = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeed();
+        int position = getPosition(key);
+        Node<K, V> node = storage[position];
+        if (node == null) {
+            storage[position] = new Node<>(key, value);
+            size++;
+        } else {
+            while (node != null) {
+                if (Objects.equals(node.key, key)) {
+                    node.value = value;
+                    return;
+                }
+                if (node.nextNode == null) {
+                    node.nextNode = new Node<>(key, value);
+                    size++;
+                    return;
+                }
+                node = node.nextNode;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = storage[getPosition(key)];
+        if (currentNode != null && Objects.equals(currentNode.key, key)) {
+            return currentNode.value;
+        } else {
```

Comment:
you can delete this block of code, cause inside while you have completely the same check

Reply:
++